pkg/controller/deployment/deployment_controller.go

```diff
@@ -493,7 +493,8 @@ func (dc *DeploymentController) updateDeploymentAndClearRollbackTo(deployment *e
 }
 
 func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Deployment) error {
-	newRS, oldRSs, err := dc.getAllReplicaSets(deployment)
+	// Don't create a new RS if not already existed, so that we avoid scaling up before scaling down
+	newRS, oldRSs, err := dc.getAllReplicaSets(deployment, false)
```

Comment:
Wouldn't it be simpler if in the place where we create the newRC, check if it's recreate, if yes, make sure spec.replicas is zero instead of passing around booleans?